src/compute-client/src/controller/instance.rs

```diff
@@ -935,6 +935,12 @@ where
             replica_write_frontier.join_assign(&since.to_owned());
         }
 
+        // If the `as_of` is empty, we are not going to create a dataflow, so don't install any read
+        // holds.
+        if as_of.is_empty() {
+            replica_write_frontier.join_assign(&Antichain::new())
```

Comment:
```suggestion
            replica_write_frontier = Antichain::new();
```

Reply:
I'd like to propose moving this the the initialization of `replica_write_frontier` further above, to have it in one place, e.g.:

```rust
let mut replica_write_frontier = if as_of.is_empty() {
    // If the `as_of` is empty, we are not going to create a dataflow, so we initialize the 
    // per-replica frontiers to the empty frontier as well, to communicate that we don't
    // expect frontier updates.
    Antichain::new()
} else {
    // Otherwise, when we initialize per-replica write frontiers (and thereby the per-replica
    // read capabilities), we cannot use the `as_of` because of reconciliation: Existing
    // slow replicas might be reading from the inputs at times before the `as_of` and we
    // would rather not crash them by allowing their inputs to compact too far. So instead
    // we initialize the per-replica write frontiers with the smallest possible value that
    // is a valid read capability for all inputs, which is the join of all input `since`s.
    Antichain::from_elem(T::minimum());
};
```

Reply:
Note that `replica_write_frontier` is modified twice after its initialization. Currently these are just `join_assign`s, so initializing it to `[]` beforehand would be ok, but it still seems a bit dangerous: we might add some code do something else to it, and then the `[]` initialization might get lost.

The current place where I have the change to `[]` is after all other modifications (but before any use), which looks like a safer place for it. Or should I maybe do a bigger restructure where the 2 loops that have the `replica_write_frontier.join_assign` would be moved into a new block, whose result would be assigned as the initial value of `replica_write_frontier`?

Reply:
> but it still seems a bit dangerous: we might add some code do something else to it, and then the `[]` initialization might get lost.

I would hope that people read and understand all that code before making changes to it, but you might be right that we maybe shouldn't rely on that... Fine to leave it as it then!